metrics/bleu/README.md

```diff
@@ -2,35 +2,37 @@
+BLEU (Bilingual Evaluation Understudy) is an algorithm for evaluating the quality of text which has been machine-translated from one natural language to another. Quality is considered to be the correspondence between a machine's output and that of a human: "the closer a machine translation is to a professional human translation, the better it is" – this is the central idea behind BLEU. BLEU was one of the first metrics to claim a high correlation with human judgements of quality, and remains one of the most popular automated and inexpensive metrics.
 
-Scores are calculated for individual translated segments—generally sentences—by comparing them with a set of good quality reference translations. Those scores are then averaged over the whole corpus to reach an estimate of the translation's overall quality. Neither intelligibility nor grammatical correctness are not taken into account. 
+Scores are calculated for individual translated segments—generally sentences—by comparing them with a set of good quality reference translations. Those scores are then averaged over the whole corpus to reach an estimate of the translation's overall quality. Neither intelligibility nor grammatical correctness are not taken into account.
 
 ## Intended Uses
 BLEU and BLEU-derived metrics are most often used for machine translation.
 
 ## How to Use
 
-This metric takes as input lists of predicted sentences and reference sentences:
+This metric takes as input a list of predicted sentences and a list of lists of reference sentences (since each predicted sentence can have multiple references):
 
 ```python
->>> predictions = [
-...     ["hello", "there", "general", "kenobi",
-...     ["foo", "bar" "foobar"]
-... ]
+>>> predictions = ["hello there general kenobi", "foo bar foobar"]
 >>> references = [
-...     [["hello", "there", "general", "kenobi"]],
-...     [["foo", "bar", "foobar"]]
+...     ["hello there general kenobi", "hello there !"],
+...     ["foo bar foobar"]
 ... ]
 >>> bleu = evaluate.load_metric("bleu")
 >>> results = bleu.compute(predictions=predictions, references=references)
 >>> print(results)
-{'bleu': 0.6370964381207871, 'precisions': [0.8333333333333334, 0.75, 1.0, 1.0], 'brevity_penalty': 0.7165313105737893, 'length_ratio': 0.75, 'translation_length': 6, 'reference_length': 8}
+{'bleu': 1.0, 'precisions': [1.0, 1.0, 1.0, 1.0], 'brevity_penalty': 1.0, 'length_ratio': 1.1666666666666667, 'translation_length': 7, 'reference_length': 6}
```

Comment:
Any idea why the results are not the same ?

Reply:
that's a good question. I would guess it's the tokenizer, I tried running the original version (from `datasets`) and this version, and they do give different results:
`datasets` version of BLEU:
```
>>> from datasets import load_metric
>>> bleu = load_metric('bleu')
>>> predictions=[["hello", "there", "general", "kenobi"],["foo", "bar" "foobar"]]
>>> references= [[["hello", "there", "general", "kenobi"]],[["foo", "bar", "foobar"]]]
>>> bleu.compute(predictions=predictions, references=references)
{'bleu': 0.7526405111736054, 'precisions': [0.8333333333333334, 0.75, 1.0, 1.0], 'brevity_penalty': 0.846481724890614, 'length_ratio': 0.8571428571428571, 'translation_length': 6, 'reference_length': 7}
```
New version:

```
>>> from evaluate import load_metric
>>> import evaluate
>>> bleu2 = load_metric('/home/sasha/Documents/HuggingFace/evaluate/metrics/bleu')
>>> predictions2 = ["hello there general kenobi", "foo bar foobar"]
>>> references2 = [["hello there general kenobi", "hello there !"],["foo bar foobar"]]
>>> bleu2.compute(predictions=predictions2, references=references2)
{'bleu': 1.0, 'precisions': [1.0, 1.0, 1.0, 1.0], 'brevity_penalty': 1.0, 'length_ratio': 1.1666666666666667, 'translation_length': 7, 'reference_length': 6}
```
What do you think @lhoestq ? Maybe it's literally the `!` that gets tokenized differently?...

Reply:
I added a second `reference` compared to the previous example (to illustrate the functionality).
I triple-checked though, and the results are the same for the previous version and the current version of the code! :hugs: